### PR TITLE
chore: add showcase testing for common service mixins (location)

### PR DESCRIPTION
**Pull request by mpeddada1**
**Notable Changes**

- Modified BUILD.bazel file to add location.proto as a dependency.
- Regenerated clients with `mvn compile -Pupdate` after modification of BUILD.bazel.
- Added ITCommonServiceMixins showcase integration tests.

### CODE REVIEW

File: showcase/BUILD.bazel [hunk 1]
```diff
@@ -19,14 +19,16 @@ proto_library_with_info(
...
     ],
 )
 
 java_proto_library(
     name = "showcase_java_proto",
     deps = [
         "@com_google_gapic_showcase//schema/google/showcase/v1beta1:showcase_proto",
-#        "//showcase/gapic-showcase-extended/proto:showcase_proto_extended",
+#       "//showcase/gapic-showcase-extended/proto:showcase_proto_extended",
+        "@com_google_googleapis//google/cloud/location:location_proto"
```
Comment (lqiu96) [inline showcase/BUILD.bazel:31]: Wondering if this is needed in the bazel target. Looking at an example bazel rule in googleapis with location mixin doesn't seem to include it in `java_proto_library`: https://github.com/googleapis/googleapis/blob/a32790cd647c34be01845e0edd580557bced3d42/google/cloud/tpu/v2/BUILD.bazel#L59-L62

File: showcase/BUILD.bazel [hunk 2]
```diff
@@ -56,11 +58,14 @@ java_gapic_library(
     test_deps = [
         ":showcase_java_grpc",
 #        ":showcase_java_grpc_extended",
+         "@com_google_api_grpc_proto_google_common_protos//google/cloud/location:location_java_grpc",
     ],
     transport = "grpc+rest",
     deps = [
         ":showcase_java_proto",
         "@com_google_googleapis//google/api:api_java_proto",
+        "@com_google_api_grpc_proto_google_common_protos//:com_google_api_grpc_proto_google_common_protos",
```
Comment (lqiu96) [inline showcase/BUILD.bazel:67]: Same with this bazel dep

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCommonServiceMixins.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    assertThat(location)
+        .isEqualTo(
+            Location.newBuilder()
+                .setName("projects/showcase/location/us-central1")
+                .setDisplayName("us-central1")
+                .build());
+  }
+
+  @Test
+  public void testGrpc_listLocations() {
```
Comment (lqiu96) [inline showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCommonServiceMixins.java:66]: Might be helpful to add a comment that references the gapic-showcase implementation for the location service: https://github.com/googleapis/gapic-showcase/blob/main/server/services/locations_service.go#L35-L78

I was wondering how these values were populated/ returned back.

File: showcase/BUILD.bazel (hunk 1 above)
Comment (mpeddada1) [reply showcase/BUILD.bazel:31]: Thanks for pointing this out! You're right, looks like it is not needed. I will remove it.

File: showcase/BUILD.bazel (hunk 2 above)
Comment (mpeddada1) [reply showcase/BUILD.bazel]: Hm  I tried removing this dep but it appears to result in the following error:

```
(14:11:07) ERROR: /Users/mpeddada/IdeaProjects2/gapic-generator-java/showcase/BUILD.bazel:48:19: Building showcase/libshowcase_java_gapic.jar (1 source jar) failed: (Exit 1): java failed: error executing command external/remotejdk11_macos/bin/java -XX:-CompactStrings '--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED' '--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED' ... (remaining 17 arguments skipped)
bazel-out/darwin-fastbuild/bin/showcase/_javac/showcase_java_gapic/libshowcase_java_gapic_tmp/com/google/showcase/v1beta1/EchoClient.java:834: error: [strict] Using type com.google.cloud.location.ListLocationsRequest from an indirect dependency (TOOL_INFO: "@com_google_api_grpc_proto_google_common_protos//:com_google_api_grpc_proto_google_common_protos"). See command below **
  public final ListLocationsPagedResponse listLocations(ListLocationsRequest request) {
                                                        ^
 ** Please add the following dependencies: 
  @com_google_api_grpc_proto_google_common_protos//:com_google_api_grpc_proto_google_common_protos to //showcase:showcase_java_gapic 
 ** You can use the following buildozer command: 
buildozer 'add deps @com_google_api_grpc_proto_google_common_protos//:com_google_api_grpc_proto_google_common_protos' //showcase:showcase_java_gapic 

Target //showcase:update_gapic failed to build
Use --verbose_failures to see the command lines of failed build steps.
(14:11:07) INFO: Elapsed time: 138.978s, Critical Path: 36.58s
(14:11:07) INFO: 247 processes: 10 internal, 221 darwin-sandbox, 16 worker.
(14:11:07) FAILED: Build did NOT complete successfully
(14:11:07) ERROR: Build failed. Not running target
(14:11:07) FAILED: Build did NOT complete successfully

```

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCommonServiceMixins.java (hunk 3 above)
Comment (mpeddada1) [reply showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITCommonServiceMixins.java:66]: Good call, done.

File: showcase/BUILD.bazel (hunk 1 above)
Comment (mpeddada1) [reply showcase/BUILD.bazel]: Hm surprisingly, also removing these two deps (https://github.com/googleapis/googleapis/blob/a32790cd647c34be01845e0edd580557bced3d42/google/cloud/tpu/v2/BUILD.bazel#L79-L85) didn't seem to make a difference to the client re-generation.

**Review by burkedavison — APPROVED** [review]

File: showcase/BUILD.bazel (hunk 1 above)
Comment (lqiu96) [reply showcase/BUILD.bazel]: Ah interesting! Let me try locally

File: showcase/BUILD.bazel (hunk 2 above)
Comment (lqiu96) [reply showcase/BUILD.bazel]: Ah let me try this locally

Comment (lqiu96) [reply showcase/BUILD.bazel]: Tried it with:
```
java_gapic_library(
    name = "showcase_java_gapic",
    srcs = [":showcase_proto_with_info"],
    gapic_yaml = None,
    grpc_service_config = "@com_google_gapic_showcase//schema/google/showcase/v1beta1:showcase_grpc_service_config.json",
    # TODO(#1285): Enable rest_numeric_enums once https://github.com/googleapis/gapic-showcase/issues/1255 is
    # fixed.
    rest_numeric_enums = False,
    service_yaml = "@com_google_gapic_showcase//schema/google/showcase/v1beta1:showcase_v1beta1.yaml",
    test_deps = [
        ":showcase_java_grpc",
#        ":showcase_java_grpc_extended",
        "@com_google_googleapis//google/cloud/location:location_java_grpc",
    ],
    transport = "grpc+rest",
    deps = [
        ":showcase_java_proto",
        "@com_google_googleapis//google/api:api_java_proto",
        "@com_google_googleapis//google/cloud/location:location_java_proto",
   ],
)
```
and `mvn verify -P enable-integration-tests` works. I'm guessing it's that `@com_google_api_grpc_proto_google_common_protos//:com_google_api_grpc_proto_google_common_protos` and `"@com_google_googleapis//google/cloud/location:location_java_proto"` both contain the location dep?

Not sure which one is better...

**Review by lqiu96 — APPROVED** [review]

**Pull request merged**